src/ripple/rpc/handlers/LedgerData.cpp

```diff
@@ -72,34 +67,37 @@ Json::Value doLedgerData (RPC::Context& context)
         limit = jLimit.asInt ();
     }
 
+    auto maxLimit = RPC::Tuning::pageLength(isBinary);
     if ((limit < 0) || ((limit > maxLimit) && (context.role != Role::ADMIN)))
         limit = maxLimit;
 
     jvResult[jss::ledger_hash] = to_string (lpLedger->info().hash);
-    jvResult[jss::ledger_index] = std::to_string(lpLedger->info().seq);
+    jvResult[jss::ledger_index] = lpLedger->info().seq;
 
-    Json::Value& nodes = (jvResult[jss::state] = Json::arrayValue);
+    Json::Value& nodes = jvResult[jss::state];
 
-    for(auto const& sle : lpLedger->sles)
+    while ((key = lpLedger->succ(*key)))
```

Comment:
I don't know how much of a win is there but kind of surprised that we never came up with a stateful iterator that took an `after` param (and maybe an optional `before`). It could be used here and for walking all qualities of a given order book base.